aws/src/main/java/org/apache/iceberg/aws/s3/S3InputStream.java

```diff
@@ -194,9 +195,29 @@ private void openStream() throws IOException {
     }
   }
 
-  private void closeStream() throws IOException {
+  private void closeStream() {
     if (stream != null) {
-      stream.close();
+      // if we aren't at the end of the stream, and the stream is abortable, then
+      // call abort() so we don't read the remaining data with the Apache HTTP client
+      abortStream();
+      try {
+        stream.close();
+      } catch (Exception e) {
+        // log at trace level as closing an aborted stream will throw a content length
+        // check exception with the Apache HTTP client
+        LOG.trace("Error closing stream", e);
```

Comment:
That means we will see exceptions for every stream close if trace is enabled. If it is expected to fail with the content length check, can we at least skip that? Then we don't need to log in trace, but we can log in maybe a warning level.

Reply:
@singhpk234 @danielcweeks are you ok if I revert this back to a no-op?

Reply:
I guess the question to ask before this is, why do we need to close it again if it's already aborted? If you see in the Trino logic I referenced, it's an if-else.

Reply:
That would work for the Apache HTTP client, the close doesn't do anything for that. But aborting the stream opened by URL connection client doesn't do anything, so we need to close it in that case.

Reply:
I see, that's interesting difference, thanks for the explanation!